### ISSUE TITLE
Stuck using Qt Multimedia

### ISSUE THREAD

**Issue by sergiobenrocha2** [mgba-emu/mgba#7]
Change the Audio Driver to Qt Multimedia and try to load any rom, it will stuck with a white screen.

In the second time you try to load a rom, you'll see this message:

```
using null output device, none available
```

Using qtmultimedia5-dev 5.2.1, xubuntu 14.04 amd64.

**Comment by sergiobenrocha2** [comment]
It only works if you disable audio sync (working with no audio).

**Comment by purpasmart96** [comment]
same thing happens on windows

**Comment by endrift** [comment]
On Linux, I think this has to do with whether or not PulseAudio is running.

On Windows, I keep getting conflicting reports about whether or not this is working. @purpasmart96, which nightly are you running? It's (supposedly) fixed as of the nightly from two days ago. Can you try with the latest nightly?

**Comment by purpasmart96** [comment]
@endrift Yeah, it now works for windows now, but it wasn't a for a while.

**Comment by endrift** [comment]
@purpasmart96 Are you still encountering this issue with the latest nightlies?

**Comment by purpasmart96** [comment]
@endrift Haven't tested on the latest nightly, will test soon.

**Comment by sergiobenrocha2** [comment]
Sadly ubuntu doesn't provide debug packages for libQt5Core

```
mgba-qt: /home/sergio/Projects/mGBA/mgba-git/src/third-party/blip_buf/blip_buf.c:144: blip_set_rates: Assertion `0 <= factor - m->factor && factor - m->factor < 1' failed.

Program received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffd67f4700 (LWP 2902)]
0x00007ffff4f5ecc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56	../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt full
#0  0x00007ffff4f5ecc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
        resultvar = 0
        pid = 2900
        selftid = 2902
#1  0x00007ffff4f620d8 in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x7fffffffe30b, sa_sigaction = 0x7fffffffe30b}, sa_mask = {__val = {140737304481052, 140737346418832, 144, 4294967295, 
              140737303123171, 4294967296, 140736792049200, 12742143496, 140736792049120, 140737488350644, 0, 0, 0, 21474836480, 140737352028160, 140737304496176}}, 
          sa_flags = -141936424, sa_restorer = 0x7ffff78a3a33 <__PRETTY_FUNCTION__.2935>}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x00007ffff4f57b86 in __assert_fail_base (fmt=0x7ffff50a8830 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x7ffff78a38d8 "0 <= factor - m->factor && factor - m->factor < 1", 
    file=file@entry=0x7ffff78a3890 "/home/sergio/Projects/mGBA/mgba-git/src/third-party/blip_buf/blip_buf.c", line=line@entry=144, 
    function=function@entry=0x7ffff78a3a33 <__PRETTY_FUNCTION__.2935> "blip_set_rates") at assert.c:92
        str = 0x7fffd00039c0 "x"
        total = 4096
#3  0x00007ffff4f57c32 in __GI___assert_fail (assertion=assertion@entry=0x7ffff78a38d8 "0 <= factor - m->factor && factor - m->factor < 1", 
    file=file@entry=0x7ffff78a3890 "/home/sergio/Projects/mGBA/mgba-git/src/third-party/blip_buf/blip_buf.c", line=line@entry=144, 
    function=function@entry=0x7ffff78a3a33 <__PRETTY_FUNCTION__.2935> "blip_set_rates") at assert.c:101
No locals.
#4  0x00007ffff788a4c3 in blip_set_rates (m=<optimized out>, clock_rate=<optimized out>, sample_rate=<optimized out>)
    at /home/sergio/Projects/mGBA/mgba-git/src/third-party/blip_buf/blip_buf.c:144
        factor = <optimized out>
        __PRETTY_FUNCTION__ = "blip_set_rates"
#5  0x00000000004ae374 in QGBA::AudioDevice::setFormat (this=0x7fffd0003150, format=...) at /home/sergio/Projects/mGBA/mgba-git/src/platform/qt/AudioDevice.cpp:39
        fauxClock = <optimized out>
#6  0x00000000004adc84 in QGBA::AudioProcessorQt::start (this=0x9ed4c0) at /home/sergio/Projects/mGBA/mgba-git/src/platform/qt/AudioProcessorQt.cpp:61
No locals.
#7  0x00007ffff5da622e in QObject::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#8  0x00007ffff6e0dc8c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
No symbol table info available.
#9  0x00007ffff6e12e56 in QApplication::notify(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
No symbol table info available.
#10 0x00007ffff5d7dc2d in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#11 0x00007ffff5d7fe07 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#12 0x00007ffff5dcacd3 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#13 0x00007fffefd14e04 in g_main_dispatch (context=0x7fffd0000990) at /build/buildd/glib2.0-2.40.2/./glib/gmain.c:3064
        dispatch = 0x7ffff5dcacc0
        prev_source = 0x0
        was_in_call = 0
        user_data = 0x0
---Type <return> to continue, or q <return> to quit---
        callback = 0x0
        cb_funcs = <optimized out>
        cb_data = 0x0
        need_destroy = <optimized out>
        source = 0x7fffd0001340
        current = 0x92c480
        i = 0
#14 g_main_context_dispatch (context=context@entry=0x7fffd0000990) at /build/buildd/glib2.0-2.40.2/./glib/gmain.c:3663
No locals.
#15 0x00007fffefd15048 in g_main_context_iterate (context=context@entry=0x7fffd0000990, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>)
    at /build/buildd/glib2.0-2.40.2/./glib/gmain.c:3734
        max_priority = 2147483647
        timeout = -1
        some_ready = 1
        nfds = <optimized out>
        allocated_nfds = 1
        fds = 0x7fffd00013c0
#16 0x00007fffefd150ec in g_main_context_iteration (context=0x7fffd0000990, may_block=1) at /build/buildd/glib2.0-2.40.2/./glib/gmain.c:3795
        retval = <optimized out>
#17 0x00007ffff5dca98c in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#18 0x00007ffff5d7c96b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#19 0x00007ffff5ba523a in QThread::exec() () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#20 0x00007ffff5ba9233 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#21 0x00007ffff2a08182 in start_thread (arg=0x7fffd67f4700) at pthread_create.c:312
        __res = <optimized out>
        pd = 0x7fffd67f4700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140736792053504, 830988775011140885, 1, 0, 140736792054208, 140736792053504, -830932697753828075, -831012390027246315}, 
              mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
        __PRETTY_FUNCTION__ = "start_thread"
#22 0x00007ffff502247d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
No locals.
```

**Comment by vgturtle127** [comment]
Are you going to just switch to SDL for audio or just get rid of Qt multimedia, or?

I mean, Qt seems like it has some pretty numerous issues.

**Comment by endrift** [comment]
I already did that, but I'd like to try to fix it, which is why I left the bug open.

**Comment by vgturtle127** [comment]
@endrift You mean you still have left the option to use Qt multimedia, but the default is SDL. And you are just wanting to fix the issues with Qt multimedia?

**Comment by endrift** [comment]
Qt multimedia is disabled in the Windows builds that are distributed entirely. I'd like to fix the issues at some point but it's very low priority.

**Comment by vgturtle127** [comment]
@endrift So you mentioned Windows and Linux, but does the problem with Qt multimedia persist on Mac? I have never used Qt to build on Mac, but I assumed if you were using Qt multimedia on Linux and Windows you may use it on Mac.

**Comment by endrift** [comment]
It's not great on OS X but it works, unlike Windows.

**Comment by vgturtle127** [comment]
I've always found Mac development harder. :disappointed: 

I wonder if Qt knows about this?

**Comment by endrift** [comment]
Shoooould be fixed in 24270af I hope.

**Comment by endrift** [comment]
Closing due to lack of reply.
